File: backend/src/main/java/com/bookloop/api/transaction/Transaction.java
package com.bookloop.api.transaction;

import com.bookloop.api.book.Book;
import com.bookloop.api.match.Match;
import com.bookloop.api.user.User;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "transactions")
public class Transaction {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Enumerated(EnumType.STRING)
    @NotNull
    private TransactionStatus status;

    @NotNull
    @ManyToOne
    @JoinColumn(name = "match_id")
    private Match match;

    @NotNull
    @ManyToOne
    @JoinColumn(name = "book_id")
    private Book book;

    @NotNull
    @ManyToOne
    @JoinColumn(name = "proponent_id")
    private User proponent;

    @NotNull
    @ManyToOne
    @JoinColumn(name = "requester_id")
    private User requester;

    @CreationTimestamp
    private LocalDateTime createdAt;
}

File: mobile/models/Transaction.ts
export type TransactionStatus= 
| "PENDENTE"
| "FINALIZADA"
| "CANCELADA";

export type TransactionType=
| "TROCA"
| "DOACAO"

export interface Transaction{
    id: number;
    status: TransactionStatus;
    matchId: number | null;
    bookId: number;
    bookTitle: string;
    proponentId: number;
    proponentName: string;
    requesterId: number;
    requesterName: string;
    createdAt: string | null;
    type: TransactionType;
    otherUserId: number;
    otherUserName: string;
}